client/components/homepage/Login.jsx

```diff
@@ -15,26 +13,40 @@ class Login extends React.Component {
               className="col-md-12"
               type="text"
               placeholder="Enter a username"
+              onChange={(e) => { props.addUserName(e.target.value); }}
             />
             <input
               className="col-md-12"
               type="text"
               placeholder="Enter an e-mail address"
+              onChange={(e) => { props.addUserEmail(e.target.value); }}
             />
-            <input
-              className="col-md-12"
-              type="text"
-              placeholder="Enter a password"
-            />
-            <button>Sign me up!</button>
+            <Link to="/">Sign me up!</Link>
           </form>
-          <h3 className="text-center">Or...</h3>
-          <button>Sign in with google button</button>
-          <button>Sign in with facebook button</button>
         </div>
       </div>
     );
-  }
-}
+};
+
+
+const mapStateToProps = state => ({
+  userName: state.userName,
+  email: state.email
+});
+
+const mapDispatchToProps = dispatch => ({
+  addUserName: name => dispatch(addUserName(name)),
+  addUserEmail: email => dispatch(addUserEmail(email))
+});
+
+export { Login };
+export default connect(mapStateToProps, mapDispatchToProps)(Login);
```

Comment:
Maybe leave a comment explaining that we're exporting the component in two ways, one for testing { Login } and one for use in the app.